yb-voyager/src/srcdb/postgres.go

```diff
@@ -180,3 +190,33 @@ func (pg *PostgreSQL) ExportDataPostProcessing(exportDir string, tablesProgressM
 	}
 	dfd.Save()
 }
+
+//find correct path of pg command depending on the version required
+func FindCorrectPGCommandPath(pathVariable *string, executableName string) {
+	paths, err := findAllExecutablesInPath(executableName)
+	if err != nil {
+		utils.ErrExit("error in finding executables: %v", err)
+	}
+
+	for _, path := range paths {
+		cmd := exec.Command(path, "--version")
+		stdout, err := cmd.Output()
+		if err != nil {
+			utils.ErrExit("error in finding version of %v from path %v: %v", executableName, path, err)
+		}
+
+		//example output: pg_restore (PostgreSQL) 14.5
+		version, err := strconv.ParseFloat(regexp.MustCompile(`[ \n]`).Split(string(stdout), -1)[2], 64)
```

Comment:
Minor nit, but this line has low readability, you can consider using a regex match similar to: 
`(?i)pg_[\S]+ \(PostgreSQL\) ([0-9.]+)`
With the help of `pg_regexp.FindStringSubmatch(string(stdout))`, the first group this returns (if at all) would be the version in its entirety.
Feel free to change the regex to be more generic if you see the possibility, I just came up with this from the top of my head

Reply:
I used the code provided by @amit-yb here